### ISSUE TITLE
3D lattice graph fails to be created

### ISSUE THREAD

**Issue by josemduarte** [eppic-team/eppic#159]
After the contraction refactoring (PR #154), the 3D lattice graph in NGL got broken.

The issue is in `isIsologous` which now is taken from `StructureInterface` in the initialisation of the LatticeGraph. Since the `StructureInterface` is created by conversion from a `InterfaceDB` it misses some data and so an NPE occurs.

Stack trace:
```
java.util.concurrent.ExecutionException: java.lang.NullPointerException
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[na:1.8.0_111]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[na:1.8.0_111]
        at ch.systemsx.sybit.crkwebui.server.commons.util.io.FileCache$CacheFile.lambda$new$1(FileCache.java:86) ~[classes/:na]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_111]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_111]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_111]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_111]
        at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_111]
Caused by: java.lang.NullPointerException: null
        at org.biojava.nbio.structure.contact.StructureInterface.getContactOverlapScore(StructureInterface.java:597) ~[biojava-structure-4.2.5.jar:4.2.5]
        at org.biojava.nbio.structure.contact.StructureInterface.isIsologous(StructureInterface.java:659) ~[biojava-structure-4.2.5.jar:4.2.5]
        at eppic.assembly.InterfaceEdge.setInterface(InterfaceEdge.java:84) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at eppic.assembly.LatticeGraph.initLatticeGraphTopologically(LatticeGraph.java:337) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at eppic.assembly.LatticeGraph.<init>(LatticeGraph.java:116) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at eppic.assembly.LatticeGraph3D.<init>(LatticeGraph3D.java:77) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at ch.systemsx.sybit.crkwebui.server.jmol.generators.LatticeGraphPageGenerator.lambda$generateJSONPage$0(LatticeGraphPageGenerator.java:141) ~[classes/:na]
        ... 4 common frames omitted
22:28:00.799 [qtp186276003-355] ERROR c.s.s.c.s.j.s.LatticeGraphServlet - Error during preparation of Lattice Graph page.
java.util.concurrent.ExecutionException: java.lang.NullPointerException
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[na:1.8.0_111]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[na:1.8.0_111]
        at ch.systemsx.sybit.crkwebui.server.commons.util.io.FileCache$CacheFile.get(FileCache.java:111) ~[classes/:na]
        at ch.systemsx.sybit.crkwebui.server.commons.util.io.FileCache.getString(FileCache.java:299) ~[classes/:na]
        at ch.systemsx.sybit.crkwebui.server.jmol.generators.LatticeGraphPageGenerator.generateJSONPage(LatticeGraphPageGenerator.java:154) ~[classes/:na]
        at ch.systemsx.sybit.crkwebui.server.jmol.servlets.LatticeGraphServlet.doGet(LatticeGraphServlet.java:155) ~[classes/:na]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:687) [servlet-api-3.1.jar:3.1.0]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [servlet-api-3.1.jar:3.1.0]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808) [jetty-servlet-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669) [jetty-servlet-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:256) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:219) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.12.v20150709.jar:9.2.12.v20150709]
        at ch.systemsx.sybit.crkwebui.server.crawler.servlets.CrawlerFilter.doFilter(CrawlerFilter.java:58) [classes/:na]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:364) [jetty-servlets-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652) [jetty-servlet-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585) [jetty-servlet-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577) [jetty-security-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515) [jetty-servlet-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:110) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.12.v20150709.jar:9.2.12.v20150709]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.12.v20150709.jar:9.2.12.v20150709]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_111]
Caused by: java.lang.NullPointerException: null
        at org.biojava.nbio.structure.contact.StructureInterface.getContactOverlapScore(StructureInterface.java:597) ~[biojava-structure-4.2.5.jar:4.2.5]
        at org.biojava.nbio.structure.contact.StructureInterface.isIsologous(StructureInterface.java:659) ~[biojava-structure-4.2.5.jar:4.2.5]
        at eppic.assembly.InterfaceEdge.setInterface(InterfaceEdge.java:84) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at eppic.assembly.LatticeGraph.initLatticeGraphTopologically(LatticeGraph.java:337) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at eppic.assembly.LatticeGraph.<init>(LatticeGraph.java:116) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at eppic.assembly.LatticeGraph3D.<init>(LatticeGraph3D.java:77) ~[eppic-cli-3.0-SNAPSHOT.jar:na]
        at ch.systemsx.sybit.crkwebui.server.jmol.generators.LatticeGraphPageGenerator.lambda$generateJSONPage$0(LatticeGraphPageGenerator.java:141) ~[classes/:na]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_111]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_111]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_111]

```

**Comment by josemduarte** [comment]
Ok it turns out that we are not generating the json 3D assembly diagram file yet from CLI. The above only happens if the json file is not there. I'll implement the generation of the json 3D diagram file from CLI as a solution to the issue.

**Comment by josemduarte** [comment]
New solution implemented (3e0ea6a) that allows for any set of interfaces to work, not only those sets that are precalculated with CLI. The solution is not very efficient (requires re-calculation of interfaces) but works for this purpose.
